### PR TITLE
Reorder test steps

### PR DESCRIPTION
**Pull request by nolancon**
**TLDR**: Weird test scenario was causing some flakes on deletion, reordering fixes it.

**Full Explanation:**
Sometimes on the "delete cluster" step, the operation hangs and ultimately fails because a finalizer cannot be removed from the storagoeoscluster object.

Basically, when we upscale the cluster (ie add a second node) in the tests, a second node daemon is created, as is a second metrics-exporter daemon. However, both of these daemons will be stuck in pending because the second node is just an empty node object used to trigger the creation of the daemons. 
This is fine, but the ReadyCheck for the metrics-exporter compares 
`metricsDS.Status.NumberReady == nodeDS.Status.DesiredNumberScheduled` 
to establish readiness of the metrics-exported operand:
https://github.com/storageos/operator/blob/42d8bfddecf25850b458662b07d521ac52179bd4/controllers/storageoscluster/metrics-exporter_operand.go#L105 

(This the correct comparison to make AFAICT because a metrics-exporter should still be on a node where a node daemon has gone down).

But because the nodeDS desired number will be 2, and the metricsDS ready number can only ever be 1 in this scenario (due to dummy node), then the metrics-exporter will never be ready - as such, the cluster object cannot be declared ready either.
If the next step is to then delete the cluster object (while it is still trying to update itself), then we get this finalizer situation and a hanging cluster.

So the fix is to disable the metrics exporter before the deletion to avoid the situation altogether. 

Note: just deleting a cluster like this isn't recommended, doing `uninstall` with the plugin would handle the finalizers gracefully, so this isn't a bug or something we need to worry about - just an unfortunate test scenario.

### CODE REVIEW

**Review by Mojachieee — APPROVED** [review]
I bet this was a nightmare to debug. Nice find!
